### DISCUSSION TITLE
New version of PyRanges (breaking backwards compatibility)

### DISCUSSION THREAD

**Discussion by endrebak** [biocore-ntnu/pyranges#321] [Ideas]
I am open to majorly breaking backwards compatibility. Would warrant a discussion. 

Perhaps we could make a new object and call them genomicranges. Or a new library. This means all existing code would continue working and we could perform a major rewrite.

Of course, most of the gnarly code dealing with overlaps and so on could be reused without any problem.

If so, we should make a detailed plan and think hard about design beforehand.

**Comment by marco-mariotti** [comment]
Moved discussion to https://github.com/pyranges/pyranges/discussions/357
